src/Workspaces/Core/Portable/Shared/Extensions/ISolutionExtensions.cs

```diff
@@ -48,19 +49,22 @@ public static IEnumerable<DocumentId> GetChangedDocuments(this Solution newSolut
 
         public static TextDocument GetTextDocument(this Solution solution, DocumentId documentId)
         {
-            return solution.GetDocument(documentId) ?? solution.GetAdditionalDocument(documentId);
+            return solution.GetDocument(documentId) ?? solution.GetAdditionalDocument(documentId) ?? solution.GetAnalyzerConfigDocument(documentId);
         }
 
         public static Solution WithTextDocumentText(this Solution solution, DocumentId documentId, SourceText text, PreservationMode mode = PreservationMode.PreserveIdentity)
         {
-            var document = solution.GetTextDocument(documentId);
-            if (document is Document)
+            var textDocument = solution.GetTextDocument(documentId);
+            switch (textDocument)
             {
-                return solution.WithDocumentText(documentId, text, mode);
-            }
-            else
-            {
-                return solution.WithAdditionalDocumentText(documentId, text, mode);
+                case Document _:
+                    return solution.WithDocumentText(documentId, text, mode);
+
+                case AnalyzerConfigDocument _:
+                    return solution.WithAnalyzerConfigDocumentText(documentId, text, mode);
+
+                default:
+                    return solution.WithAdditionalDocumentText(documentId, text, mode);
```

Comment:
can we check explicitly for AdditionalDocs, and then throw in the default case?  this seems like a bug waitin to happen in the future whne we add something new.  

ideally, we could just get rid of this entire method.  it seems fishy to me.

Reply:
I am going to introduce a new sub-type of `TextDocument` for additional documents and handle them explicitly here and below method. I also think it would be good to introduce `TextDocumentKind` enum with 3 supported document kinds and expose it directly on `TextDocument` for clarity.

Reply:
I'd be +1 for the sub-type since that was problematic for me in other cases too. Do we want an enum or just use pattern matching if you need a switch, now that we have that?

Reply:
Sounds good - I'll hold off on the enum for now.

Reply:
I'm fine with any approach.  I think Kinds+Subtypes fit together fine, allowing people to pick the pattern they think is most appropriate.  But i have no strong opinion here.  My main concern is just that we should not "defualt" to one doc type, but should strictly enumerate things and throw when we get something we don't expect.